### ISSUE TITLE
Event Suggestions

### ISSUE THREAD

**Issue by bradduthie** [StirlingCodingClub/studyGroup#6]
**How to suggest, upvote, and volunteer to lead, events**

This issue is for event suggestions. Feel free to add suggestions as comments to this issue below. If you like one of the suggestions, add a thumbs up to it, found by clicking the + :-) in the upper right . You can also directly edit the individual suggestions using the elipses (...) and clicking 'edit' if you want to volunteer to lead an event, or comment on one of the suggested events (e.g., add some text below the suggestion).

**Comment by bradduthie** [comment]
**How to write a manuscript in Rmarkdown with GitHub (and do it faster than in Word)**

[**EVENT DONE ON 31 OCT 2018 (SEE REPOSITORY)](https://github.com/StirlingCodingClub/Manuscripts_in_Rmarkdown)

**Comment by bradduthie** [comment]
**An overview of how version control works, and how to use git in the command line**

**[[EVENT DONE ON 3 0CT 2018 (see repository)](https://github.com/StirlingCodingClub/version_control)]**

Given a comment from @mattnuttall00 on Issue #7 I'm actually tempted to do this and also present [GitKraken](https://www.gitkraken.com/) as a really useful tool and solution to getting repositories on GitHub to sync locally to your own computers. Out of habit, I've not used this much, but I just opened the (free) software for the first time in months. Here are the cool features:

- It automatically syncs with a GitHub account, and recognises your repositories. I was surprised to see that it not only found repositories I recently made on GitHub, but that it figured out where the corresponding folder was on my own desktop without me telling it.
- You can initialise a new repository *within* GitKraken anywhere on your desktop and GitHub will automatically know to initialise it (I tested this). So if you start a new folder in your desktop through GitKraken, everything will sync nicely.
- There are big push and pull buttons, and actually a pretty impressive layout for visualising everything. But it definitely solves the problem of how to connect stuff you start doing on your own computer with stuff that gets put on GitHub.
- It works with all operating systems (Windows, Mac, and Linux).

There's only one thing that didn't really work well -- I tried to fork @jejoenje 's test repository to see if it would just put it in some default location on my hard drive (it didn't -- you have to specify where it should go). 

Would people be keen on this subject, with GitKraken as an example of a quite intuitive software for keeping everything together?

**Comment by bradduthie** [comment]
**An introduction to coding in R using Rstudio**

Anna: this hasn't been very popular with upvotes, but I think it's a good idea for the next event, especially for new users
* this could / should include deciding on and setting up the SCC and semin[r] filing systems, naming conventions, workflow, tidy data ethos, etc.

**Comment by bradduthie** [comment]
**An introduction to building interactive documents in R with [shiny](https://shiny.rstudio.com/)**

**Comment by bradduthie** [comment]
**An introduction to using loops in R**

**[EVENT DONE ON 23 JAN 2019 ([SEE REPOSITORY](https://github.com/StirlingCodingClub/using_loops))]**

**Comment by bradduthie** [comment]
**How to build an individual-based model in R**

**Comment by bradduthie** [comment]
**How to build an R package**

**Comment by mattnuttall00** [comment]
**Hacky hour events**

Schedule some "Hacky hour" events as mentioned in the Mozilla introduction.  Informal meetings (preferably in a pub) where group members can meet and chat/discuss.  Good places for new group members to meet each other

**Comment by mattnuttall00** [comment]
**Temporal autocorrelation in data/time series**

Dealing with temporal autocorrelation in data/time series.  I am NOT an expert in this, and have only recently started exploring methods.  I'd be happy to facilitate a discussion session, rather than lead one.  The idea being that hopefully people with more experience than me would show up :) 

*Brad*: I think that @jeremycusack and Emma might have some expertise in this subject.

Anna: @mattnuttall00 check out this course: https://www.prstatistics.com/course/time-series-models-for-ecologists-tsme02/ .....if this sounds like what you're interested in, get in touch with Luc who might be able to get you a deal on a place

Matt: Thanks @anna-deasey !  Fortunately I only need to deal with it a little bit, so not sure investing in a full course is necessary at the moment.  I think I've got a decent enough handle on it for what I need at the moment

**Comment by rosemckeon** [comment]
**rApache - an alternative to Shiny**

For finer control over HTML and JS, rApache server setup gives you the ability to integrate R code directly within HTML markup. If you have ever wanted to display your output in a way that there's no plotting package for, this might be for you - you can build very custom web apps with rApache.

There are no new R functions to learn, but it does require familiarity with HTML and server setup.

**Comment by jejoenje** [comment]
**What editors/GUI/IDE to use for different kinds of project**

I am interested to hear what people use as -broadly- editors. Obviously RStudio, but what else? Vi? Emacs? Atom? CodeLite? What works well for you for different purposes (e.g. writing manuscripts, R/C coding?).

**Comment by rosemckeon** [comment]
**Using RStudio Server edition**

You can use RStudio from any browser with RStudio Server edition - great for working on R projects from a tablet / chromebook / any computer away from home. I've recently set this up and it's been very handy. A bit fiddly! but fun.

**Comment by rosemckeon** [comment]
**Getting started in C/C++**

I've never done any C. Would anybody else find this useful?

*Brad*: I've got some experience with this, and with [linking C to R](https://github.com/ConFooBio/gmse/blob/master/src/resource.c#L313) (though not C++). It's a tricky language to get the hang of, but the benefit of knowing it is that a lot of tasks are orders of magnitude (not an exaggeration) faster in C/C++ than in R.

**Comment by jmcvw** [comment]
__Spatial data__

Using packages `sf`/`sp` and `raster`  to create maps.

**Comment by jmcvw** [comment]
__Workflows__

In https://github.com/StirlingCodingClub/studyGroup/issues/7#issuecomment-423504448 I asked a couple of questions about how people used git. There were a couple of different responses, and I for one, can think of a few other questions, so I'm adding it here as a possible topic.

**Comment by anna-deasey** [comment]
> **Workflows**
> 
> In [#7 (comment)](https://github.com/StirlingCodingClub/studyGroup/issues/7#issuecomment-423504448) I asked a couple of questions about how people used git. There were a couple of different responses, and I for one, can think of a few other questions, so I'm adding it here as a possible topic.

How about [this](url) a basis for a semin[r] on workflows? It's one of four [guides to better science](https://www.britishecologicalsociety.org/publications/guides-to/) from BES @jmcvw @bradduthie

**Comment by rosemckeon** [comment]
**Getting to grips with SSH keys and other authentication methods**

From all the talk on [this issue](https://github.com/StirlingCodingClub/studyGroup/issues/7) it seems like this would be really useful as a practical help session. Seems like an easy place to loose people at the first hurdle.

**Comment by jmcvw** [comment]
__Functions__

How / when / why to write in functions in R.

Being able to write your own functions is dead handy.

[[**EVENT DONE ON 28 NOV 2018 (See repository)**](https://github.com/StirlingCodingClub/Funkshuns)]

**Comment by bradduthie** [comment]
**Mathematical and computational concept reviews**

Would anyone be interested in one or more semina[r]s focusing on a refresher of mathematical or computational topics of interest, with the goal being to understand a key concept in the mathematical or computational tool box? Examples might include the following:

- *What are eigenvalues and eigenvectors, and how should we think about them?*
- *What are integrals, derivatives, and partial derivatives, and why are they useful*?
- *How does matrix multiplication work, and how can we visualise matrices?*
- *How are algorithms analysed, and what is 'Big O' notation?*

The idea would be to provide a quick review on some aspect of mathematics or computer science, and ideally to present it in a slightly different way than many of us -- particularly in the life sciences -- might have been taught it in the past (with emphasis on concepts rather than calculations).

**Comment by adamaki** [comment]
@bradduthie absolutely! As a biological scientist who uses programming and maths a lot but has no advanced mathematical or computer science training (other than self-taught), I know I have some big gaps in my basic knowledge. These and any other similar topics would be a great help!
